### DISCUSSION TITLE
Logic eval error: - would result negative

### DISCUSSION THREAD

**Discussion by temptemp3** [reach-sh/reach-lang#1581] [Q&A]
I'm making a remote API call, that accesses boxes in its consensus block. There is an `Logic eval error: - would result negative` error when trying to access boxes with a map defined as `const aM = new Map(Tuple(Address, Address), UInt)` using the ALGO connector.

The line in the test script is here
https://github.com/nash-protocol/starter-kit/blob/main/index.mjs#L199

In the remote contract we have

```
  const clickM = new Map(UInt);
  const likeM = new Map(Tuple(Address, Address), UInt);
  const a = {
    U0: API("U0", {
      foo4: Fun([Address, Address], Bool),
     });
    .api_(a.U0.foo4, (addr1, addr2) => {
      return [
        (k) => {
          k(true);
          clickM[addr1] = fromSome(clickM[addr1], 0) + 1;
          clickM[addr2] = fromSome(clickM[addr2], 0) + 1;
          likeM[[addr1, addr2]] = 0;
          return [s];
        },
      ];
    })
```

In the contract that is calling the remote app we have

```
  const a = {
    master: API("Master", {
      foo4: Fun([Contract, Address, Address], Bool),
    }),
  }
      .api_(a.master.foo4, (ctc, addr1, addr2) => {
      return [
        (k) => {
          k(true);
          const r = remote(ctc, {
            U0_foo4: Fun([Address, Address], Bool),
          });
          k(
            r.U0_foo4.ALGO({
              fees: 1,
              apps: [ctc],
              boxes: [
                [ctc, 0, addr1],
                [ctc, 0, addr2],
                [ctc, 1, [addr1, addr2]],
              ],
            })(addr1, addr2)
          );
          return [];
        },
      ];
    })
```

The error occurs when trying to make a remote call to can app using a map keyed with by an address tuple of size 2.

Does anyone know what might be missing to avoid this type of error?

**Comment by jeapostrophe** [comment]
When a contract creates a box, it needs to pay for the space. The contract will check to see if its minimum balance will cover the cost. (For example, you may free box A and then allocate box B, so you don't actually need to increase your MBR.) My guess is that this remote call is creating a box and you need to include a payment to increase it.

**Reply by temptemp3** [reply]
Makes sense. It happens to be creating a box. Let me see if I can work around it by making a payment.

Btw, would that be in in `opts.fees` or REMOTE_FUNC.pay`?

REMOTE_FEE.pay because it would not be received by the remote contract account.

**Reply by jeapostrophe** [reply]
It would be a `pay` and you need to compute the correct amount. The `mbr` value returned by this function is the amount: https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/ALGO.ts#L1532

**Comment by temptemp3** [comment]
Got it. I'll give it a try. Thanks!

2023年4月20日(木) 10:49 Jay McCarthy ***@***.***>:
